### PR TITLE
Qt/SettingsWindow: Use tabs

### PR DESCRIPTION
**Pull request by spycrab**
This PR makes the settings window use tabs as to be consistent with every other dialogue in the UI.

Screenshots

### CODE REVIEW

File: Source/Core/DolphinQt2/Config/SettingsWindow.cpp [hunk 1]
```diff
@@ -21,15 +21,8 @@
...
-static int AddTab(ListTabWidget* tab_widget, const QString& label, QWidget* widget,
-                  const char* icon_name)
-{
-  int index = tab_widget->addTab(widget, label);
-  auto set_icon = [=] { tab_widget->setTabIcon(index, Resources::GetScaledThemeIcon(icon_name)); };
-  QObject::connect(&Settings::Instance(), &Settings::ThemeChanged, set_icon);
-  set_icon();
-  return index;
-}
+constexpr int GENERAL_INDEX = 0;
```
Comment (lioncash) [inline Source/Core/DolphinQt2/Config/SettingsWindow.cpp]: These be a regular enum

**Comment by spycrab** [comment]
@lioncash Updated the description.

**Comment by MayImilae** [comment]
I was going to type up a big thing justifying this in the PR, but I guess I'll do it here!

So, there are multiple reasons to change the config menu back to horizontal tabs. The first is that vertical tabs in just one menu is inconsistent with the rest of the UI, and it would take a lot of work to switch entirely to vertical tabs. Horizontal tabs is also more familiar, which will ease the WX to Qt transition. And vertical tabs only work if each icon is unique, and (for the whole UI to be vertical) would require me to make a couple dozen brand new icons and that would suck for me and delay the Qt transition considerably.

And finally... vertical tabs aren't better. When learning the UI, horizontal tabs are faster, since you can just read the name and know what they are. Once a user has learned it, due to the fixed placement of UI items, they won't read the text or look at symbols. They have just memorized the placement and are going to where they know what they want is without even paying attention to the UI items. The "symbols are better" argument comes from real life signage, when you need to recognize something very quickly in totally unfamiliar locations - like road signs. Since the locations are known, though, no one is going to be paying attention to those signs and thus that doesn't apply at all.

**Comment by Helios747** [comment]
@MayImilae your post reads like people are against this.

I feel like making the UI consistent and there being nothing wrong with horizontal tabs in the first place was a good enough reason on it's own.

**Comment by delroth** [comment]
I prefer the side list view, but meh, consistency wins.

**Comment by KAMiKAZOW** [comment]
> I prefer the side list view, but meh, consistency wins.

The alternative could have been to move the Graphics and Controller settings into the list view. ;)

**Comment by spycrab** [comment]
@KAMiKAZOW We do want to do some changes to the way our settings are spread throughout Dolphin - but at a later date.

**Pull request merged**